cli/tests/pcluster/schemas/test_schema_validators.py

```diff
@@ -553,12 +556,47 @@ def test_base_networking_validator(section_dict, expected_message):
         ("NONE", "does not match expected pattern"),
         ("subnet-12345678", None),
         ("subnet-12345678901234567", None),
+        (["subnet-12345678901234567"], "Not a valid string"),
     ],
 )
-def test_subnet_id_validator(subnet_id, expected_message):
+def test_subnet_id_validator_head_node(subnet_id, expected_message):
     """Verify that subnet ids behaves as expected when parsed in a config file."""
     _validate_and_assert_error(HeadNodeNetworkingSchema(), {"SubnetId": subnet_id}, expected_message)
-    _validate_and_assert_error(QueueNetworkingSchema(), {"SubnetIds": [subnet_id]}, expected_message)
+
+
+@pytest.mark.parametrize(
+    "subnet_ids, expected_message",
+    [
+        ([""], "does not match expected pattern"),
+        (["subnet-12345"], "does not match expected pattern"),
+        (["subnet-123456789"], "does not match expected pattern"),
+        (["NONE"], "does not match expected pattern"),
+        (["subnet-12345678"], None),
+        (["subnet-12345678901234567"], None),
+        (["subnet-1234", "subnet-5678"], "does not match expected pattern"),
+    ],
+)
+def test_subnet_id_validator_aws_batch(subnet_ids, expected_message):
+    """Verify that subnet ids behaves as expected when parsed in a config file."""
+    _validate_and_assert_error(AwsBatchQueueNetworkingSchema(), {"SubnetIds": subnet_ids}, expected_message)
+
+
+@pytest.mark.parametrize(
+    "subnet_ids, expected_message",
+    [
+        ([""], "does not match expected pattern"),
+        (["subnet-12345"], "does not match expected pattern"),
+        (["subnet-123456789"], "does not match expected pattern"),
+        (["NONE"], "does not match expected pattern"),
+        (["subnet-12345678"], None),
+        (["subnet-12345678901234567"], None),
+        (["subnet-1234", "subnet-5678"], None),
```

Comment:
Shouldn't this subnet string result in a Regex mismatch? 

RegEx ([subnet_id](https://github.com/aws/aws-parallelcluster/blob/2f1934f351f5dfdf498a3f1578c42e0de52aba38/cli/src/pcluster/schemas/common_schema.py#L34-L34)): `^subnet-[0-9a-z]{8}$|^subnet-[0-9a-z]{17}$`

Reply:
Indeed you are right. We can investigate offline.